common/src/main/java/org/opensearch/ml/common/model/LocalRegexGuardrail.java

```diff
@@ -0,0 +1,265 @@
+        builder.startObject();
+        if (stopWords != null && stopWords.size() > 0) {
+            builder.field(STOP_WORDS_FIELD, stopWords);
+        }
+        if (regex != null) {
+            builder.field(REGEX_FIELD, regex);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static LocalRegexGuardrail parse(XContentParser parser) throws IOException {
+        List<StopWords> stopWords = null;
+        String[] regex = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case STOP_WORDS_FIELD:
+                    stopWords = new ArrayList<>();
+                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        stopWords.add(StopWords.parse(parser));
+                    }
+                    break;
+                case REGEX_FIELD:
+                    regex = parser.list().toArray(new String[0]);
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+        return LocalRegexGuardrail.builder()
+                .stopWords(stopWords)
+                .regex(regex)
+                .build();
+    }
+
+    private void init() {
+        stopWordsIndicesInput = stopWordsToMap();
+        List<String> regexList = regex == null ? new ArrayList<>() : Arrays.asList(regex);
+        regexPattern = regexList.stream().map(reg -> Pattern.compile(reg)).collect(Collectors.toList());
+    }
+
+    private Map<String, List<String>> stopWordsToMap() {
+        Map<String, List<String>> map = new HashMap<>();
+        if (stopWords != null && !stopWords.isEmpty()) {
+            for (StopWords e : stopWords) {
+                if (e.getIndex() != null && e.getSourceFields() != null) {
+                    map.put(e.getIndex(), Arrays.asList(e.getSourceFields()));
+                }
+            }
+        }
+        return map;
+    }
+
+    public Boolean validateRegexList(String input, List<Pattern> regexPatterns) {
+        if (regexPatterns == null || regexPatterns.isEmpty()) {
+            return true;
+        }
+        for (Pattern pattern : regexPatterns) {
+            if (!validateRegex(input, pattern)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Boolean validateRegex(String input, Pattern pattern) {
+        Matcher matcher = pattern.matcher(input);
+        return !matcher.matches();
+    }
+
+    public Boolean validateStopWords(String input, Map<String, List<String>> stopWordsIndices) {
```

Comment:
Function name is not so easy to understand. Add some comment or change name ?

Reply:
added